o/configcore: add flag that will determine if a system option can be

alfonsosanchezbeato opened the pull request:
used on a system with modes. Use it for the kernel command line
options.

pedronis commented on overlord/configstate/configcore/runwithstate.go:
```diff
@@ -63,11 +63,11 @@ func init() {
+	addWithStateHandler(validateNetplanSettings, handleNetplanConfiguration, coreOnly)
 
 	// kernel.{,dangerous-}cmdline-append
 	// TODO we actually want this on classic with modes too
```
the todo can be removed now?

alfonsosanchezbeato replied:
Indeed, removed now

pedronis commented on overlord/configstate/configcore/kernel_test.go:
```diff
@@ -358,30 +363,43 @@ func (s *kernelSuite) testConfigureKernelCmdlineHappy(c *C, option []cmdlineOpti
 }
 
 func (s *kernelSuite) TestConfigureKernelCmdlineDangGrade(c *C) {
```
not related to the PR, but I would expand all these Dang to Dangerous

alfonsosanchezbeato replied:
Changed

pedronis commented on overlord/configstate/configcore/kernel_test.go:
```diff
@@ -421,12 +441,13 @@ func (s *kernelSuite) TestConfigureKernelCmdlineConflict(c *C) {
-	s.mockModel("signed")
+	isClassic := false
+	s.mockModelWithModeenv("signed", isClassic)
 	s.mockGadget(c)
```
shouldn't this test use core20Dev too for consistency?

alfonsosanchezbeato replied:
Good catch, changed

pedronis reviewed (commented):
https://github.com/snapcore/snapd/pull/12605/commits/d2fee0415cfe71d120fdfbb0fa2a3f02b26e3394 looks good, some small comments

alfonsosanchezbeato reviewed (commented):
Thanks, comments addressed now

mvo5 commented on tests/nested/manual/fde-on-classic/task.yaml:
```diff
@@ -111,6 +128,12 @@ execute: |
   done
 
+  # Test appending values to the kernel command line
+  append_kernel_cmdline system.kernel.cmdline-append "append.val=1 append.flag other.val=foo"
```
This is very nice

mvo5 reviewed (approved):
Thank you!

pedronis commented on overlord/configstate/configcore/kernel_test.go:
```diff
@@ -357,31 +362,44 @@ func (s *kernelSuite) testConfigureKernelCmdlineHappy(c *C, option []cmdlineOpti
 }
 
-func (s *kernelSuite) TestConfigureKernelCmdlineDangGradeDangCmdline(c *C) {
+func (s *kernelSuite) TestConfigureKernelCmdlineDangGradeDangerousCmdline(c *C) {
```
nitpick: s/DangGrade/DangerousGrade/

alfonsosanchezbeato replied:
Changed

pedronis reviewed (approved):
thank you, one nitpick

alfonsosanchezbeato commented:
Rebased now, also added an additional check in the spread test (change kernel command line, but instead of immediately rebooting, refresh the kernel).
